using-webhooks/client/android-java/app/src/main/java/com/example/app/CheckoutActivity.java

```diff
@@ -0,0 +1,170 @@
+
+import java.io.IOException;
+import java.lang.ref.WeakReference;
+import java.lang.reflect.Type;
+import java.util.Map;
+
+import android.app.Activity;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.content.Intent;
+import android.os.Bundle;
+import android.view.View;
+import android.widget.Button;
+import android.widget.Toast;
+
+import androidx.annotation.NonNull;
+import androidx.appcompat.app.AlertDialog;
+import androidx.appcompat.app.AppCompatActivity;
+
+import okhttp3.Call;
+import okhttp3.Callback;
+import okhttp3.MediaType;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.reflect.TypeToken;
+
+import com.stripe.android.ApiResultCallback;
+import com.stripe.android.PaymentIntentResult;
+import com.stripe.android.Stripe;
+import com.stripe.android.model.ConfirmPaymentIntentParams;
+import com.stripe.android.model.PaymentIntent;
+import com.stripe.android.model.PaymentMethodCreateParams;
+import com.stripe.android.view.CardInputWidget;
+
+
+public class CheckoutActivity extends AppCompatActivity {
+    /**
+     * To run this app, you'll need to first run the sample server locally.
+     * Follow the "How to run locally" instructions in the root directory's README.md to get started.
+     * Once you've started the server, open http://localhost:4242 in your browser to check that the
+     * server is running locally.
+     * After verifying the sample server is running locally, build and run the app using the
+     * Android emulator.
+     */
+    // 10.0.2.2 is the Android emulator's alias to localhost
+    private String backendUrl = "http://10.0.2.2:4242/";
+    private OkHttpClient httpClient = new OkHttpClient();
+    private String paymentIntentClientSecret;
+    private Stripe stripe;
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_checkout);
+        loadPage();
+    }
+
+    private void loadPage() {
+        // Create a PaymentIntent by calling the sample server's /create-payment-intent endpoint.
+        MediaType mediaType = MediaType.get("application/json; charset=utf-8");
+        String json = "{"
+                + "\"currency\":\"usd\","
+                + "\"items\":["
+                + "{\"id\":\"photo_subscription\"}"
+                + "]"
+                + "}";
+        RequestBody body = RequestBody.create(json, mediaType);
+        Request request = new Request.Builder()
+                .url(backendUrl + "create-payment-intent")
+                .post(body)
+                .build();
+        httpClient.newCall(request)
+                .enqueue(new Callback() {
+                    @Override
+                    public void onFailure(Call call, IOException e) {
+                        runOnUiThread(() -> {
```

Comment:
this should be running on the main thread so I don't think you need to do `runOnUiThread`